### ISSUE TITLE
Server freeze/"zombie mode"

### ISSUE THREAD

**Issue by thomasrudin** [minetest/minetest#8584]
##### Issue type
 - Bug report
 
##### Minetest version
```
5.1.0-dev (happened with 0.4.17.1 too)
568694122cb662d6c6cb071338c9d30721372b50
```

##### OS / Hardware
Operating system: Linux
CPU: x86_64

##### Summary

A couple times a week my server goes into "zombie mode":
* clients are online but the actions don't get through (no map-loading either)
* in the background (debug.txt) the loaded areas are still working (pipeworks, mesecons messages)

This can go on for minutes (about 20 minutes average)
In this time the "min-lag" (min-time between steps) increases to something above 100 ms

Monitoring image with elevated "min-lag" (yellow on the right):
<img src="https://i.imgur.com/5JXEEkw.png"></img>

I mitigated the effects of this a bit with my "watchdog" mod:
* https://git.rudin.io/minetest/watchdog/src/master/init.lua

It just measures the "min-time" between steps and if it is elevated for a while it restarts the server so everybody can play again (this is what happend in the above image).
But this should only be a temporary mitigation, i hope the root-cause can be found someday.

The server-log is filled with lots of Pathfinder messages while that happens:
```
... lots more of those below ...
2019-06-07 17:52:42: WARNING[Server]: Pathfinder: failed to update cost map
2019-06-07 17:52:42: WARNING[Server]: Pathfinder: failed to update cost map
2019-06-07 17:52:42: WARNING[Server]: Watchdog triggered!
```

Installed (relevant) mods:
* mobs_redo (monsters, animals, horse, npc)
* advtrains

Relevant links:
* monitoring code for "min-lag": https://github.com/thomasrudin-mt/monitoring/blob/master/builtin/lag.lua#L36
* Watchdog mod: https://git.rudin.io/minetest/watchdog/src/master/init.lua
* Graph: https://i.imgur.com/5JXEEkw.png (https://monitoring.minetest.land/d/YUpouLmWk/overview?orgId=1&from=1559928314152&to=1559930391076)
* My issue: https://github.com/pandorabox-io/pandorabox.io/issues/127
* Related issue: #6771

##### Steps to reproduce
* no clue, happens sporadically

**Comment by SmallJoker** [comment]
IhrFussel reported a similar issue, where the mapblocks were no longer sent to the clients until a certain player left. I believe there it's caused by either a high packet loss or wrong calculation for the mapblock re-sending delay.
However, is your case different? Does it also happen in vanilla Minetest or is the pathfinder causing that? What if you comment it out for a while?

**Comment by thomasrudin** [comment]
I just had the freeze again and checked my stats:
<img src="https://i.imgur.com/HMcAQc8.png"></img>
(The freeze took place to the right around 18:53, with the elevated min-lag)

At the same time there was a call to `minetest.find_path()` that took 7 seconds (!) to finish.
The server became unresponsive and the watchdog took it down afterwards.

@IhrFussel do you have the same issue?

**Comment by IhrFussel** [comment]
It doesn't sound like the issue I had.

Path finding is disabled in my mobs mod and I'm not sure if any other mod uses that.

**Comment by thomasrudin** [comment]
I commented out the call in `mobs_redo` to `minetest.find_path()` and
never had an issue with this anymore.

Closing my issue, as the pathfinder issue is addressed in #8642
